ignite/metrics/epoch_metric.py

```diff
@@ -68,21 +68,20 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
         if y.ndimension() == 2 and y.shape[1] == 1:
             y = y.squeeze(dim=-1)
 
-        y_pred = y_pred.to(self._predictions)
-        y = y.to(self._targets)
-
-        self._predictions = torch.cat([self._predictions, y_pred], dim=0)
-        self._targets = torch.cat([self._targets, y], dim=0)
+        self._predictions.append(y_pred.detach().to(dtype=torch.float32, device="cpu"))
+        self._targets.append(y.detach().to(dtype=torch.long, device="cpu"))
```

Comment:
No need to detach for the targets:
```suggestion
        self._targets.append(y.to(dtype=torch.long, device="cpu"))
```

Reply:
Do we need to detach() the prediction? I am afraid some model may inplace update the tensor.

Reply:
Note that .detach() does not give you new memory. If the original Tensor is modified, you will see the update.
.detach() only breaks the computational graph and prevent gradient from flowing.

If you want new memory to avoid side effects by someone modifying the other Tensor inplace, you need to use .clone().

Reply:
Yes, thanks @albanD, we need to apply `.detach().clone()` on predictions and `.clone()` on targets, such that we do not depend on any inplace updates on predictions or targets.